lib/rubocop/cop/style/redundant_regexp_quantifiers.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Checks for redundant quantifiers inside Regexp literals.
+      #
+      # @example
+      #   # bad
+      #   /(?:x+)+/
+      #
+      #   # good
+      #   /(?:x)+/
+      #
+      #   # good
+      #   /(?:x+)/
+      #
+      #   # bad
+      #   /(?:x+)?/
+      #
+      #   # good
+      #   /(?:x)*/
+      #
+      #   # good
+      #   /(?:x*)/
+      class RedundantRegexpQuantifiers < Base
+        include RangeHelp
+        extend AutoCorrector
+
+        MSG_REDUNDANT_QUANTIFIER = 'Redundant quantifiers in Regexp literal, ' \
+                                   '`%<inner_quantifier>s` and `%<outer_quantifier>s` ' \
+                                   'can be replaced with a single `%<replacement>s`.'
+
+        def on_regexp(node)
+          each_redundantly_quantified_pair(node) do |group, child|
+            replacement = merged_quantifier(group, child)
+            add_offense(
+              quantifier_range(group, child),
+              message: message(group, child, replacement)
+            ) do |corrector|
+              # drop outer quantifier
+              corrector.replace(group.loc.quantifier, '')
+              # replace inner quantifier
```

Comment:
it could also be done the other way round - replacing the outer quantifier and dropping the inner quantifier.

dropping the outer quantifier: `/(?:a+)?/ => /(?:a*)/`
dropping the inner quantifier: `/(?:a+)?/ => /(?:a)*/`

both are equivalent. i went with the first option but i don't think there's a big difference between them.